### PR TITLE
Arm backend: Rewrite static roll with slice and concatenate patterns

### PR DESCRIPTION
**Pull request by YufengShi-dudu**
Preserve supported static aten.roll nodes until the Arm backend rewrites them from:

    result = torch.roll(input, shifts=shifts, dims=dims)

to this pattern for each shift and dimension:

    shift = shift % size
    result = torch.cat(
        (
            result.slice(dim, size - shift, size),
            result.slice(dim, 0, size - shift),
        ),
        dim=dim,
    )

Apply the pattern once per nonzero shift so multi-dimensional and repeated-dimension rolls retain PyTorch semantics.

Treat roll as shared-qparam data movement so quantized inputs and outputs reuse their scale and zero point.

Leave dynamic, flattened, zero-sized, no-op, and unsupported-dtype forms on the existing fallback path.

Change-Id: I70ddb6cf77611d08c4cd1102ba59ea41f63784dc

cc @digantdesai @freddan80 @per @zingo @oscarandersson8218 @mansnils @Sebastian-Larsson @robell @rascani

### CODE REVIEW

**Review by zingo — APPROVED** [review]

**Comment by YufengShi-dudu** [comment]
The cortex-m backend failures are related. Need to investigate and fix them.

**Comment by YufengShi-dudu** [comment]
> The cortex-m backend failures are related. Need to investigate and fix them.

Resolved by a follow-up commit
